server/src/testflinger/secrets/__init__.py

```diff
@@ -44,6 +68,101 @@ def setup_secrets_store() -> SecretsStore | None:
         if not client.is_authenticated():
             raise StoreError("Vault client not authenticated")
     except requests.exceptions.ConnectionError as error:
-        raise StoreError("Unable to create Vault client") from error
+        raise StoreError("Vault client unable to connect") from error
 
     return VaultStore(client)
+
+
+def setup_mongo_store() -> MongoStore | None:
+    """
+    Set up MongoDB-based secrets store with Client-Side Field Level Encryption.
+
+    Creates a MongoStore instance using explicit CSFLE for secret storage.
+    The function handles data encryption key creation/rotation.
+
+    Returns:
+        A MongoStore, if the appropriate environment variables have been set
+        or None otherwise.
+
+    Raises:
+        StoreError: If master key is missing/invalid or encryption setup fails
+    """
+    try:
+        mongo_url = get_mongo_uri()
+    except SystemExit:
+        return None
+
+    master_key_b64 = os.environ.get("TESTFLINGER_SECRETS_MASTER_KEY")
+    if not master_key_b64:
+        return None
+
+    try:
+        master_key = base64.b64decode(master_key_b64)
+    except Exception as error:
+        raise StoreError(
+            "Environment variables with Mongo credentials are incorrect"
+        ) from error
```

Comment:
I guess this is the cleaner way to avoid catching all possible exceptions (binascii.Error, ValueError, TypeError). Can you please add a comment just mentioning this exceptions for clarity?
```suggestion
    except Exception as error:
        # binascii.Error, ValueError and TypeError can be raised during master_key decoding
        raise StoreError(
            "Environment variables with Mongo credentials are incorrect"
        ) from error
```